### PR TITLE
feat: recaptcha for contact us form

### PR DESCRIPTION
**Pull request by mrzpn**

### CODE REVIEW

**Comment by mrzpn** [comment]
wait sorry - just realised I hardcoded something while testing. will fix that first

File: src/contact/contactForm.js [hunk 1]
```diff
@@ -70,6 +107,13 @@ function ContactForm() {
         required
       />
       <br />
+      <br />
+        <ReCAPTCHA
+          ref={recaptchaRef}
+          sitekey={process.env.REACT_APP_RECAPTCHA_SITE_KEY}
+          size="invisible"
```
Comment (joanna209) [inline src/contact/contactForm.js:114]: Switch recaptcha mode from "invisible" to "I am not a robot". The latter is more secure as it provides users with an extra challenge (usually in the form of identifying X object in a set of pictures)

File: src/contact/contactForm.js [hunk 2]
```diff
@@ -1,16 +1,47 @@
...
-  const sendEmail = (e) => {
+  // configure recaptcha
+  const recaptchaRef = useRef(null)
+
+  const validateCaptcha = async () => {
+    const captchaToken = await recaptchaRef.current.executeAsync();
+    recaptchaRef.current.reset();
+
+    // pass token to backend & verify
+    try {
```
Comment (joanna209) [inline src/contact/contactForm.js:18]: Avoid using try/catch syntax and instead opt for .then()/.catch() syntax

**Review by joanna209 — CHANGES_REQUESTED** [review]

File: src/contact/contactForm.js [hunk 3]
```diff
@@ -1,16 +1,47 @@
...
+        process.env.REACT_APP_VERIFY_URL, {
+          captchaToken
+        }
+      );
+      
+      // if token verified (status 200) -> true
+      return result.status == 200 ? true : false;
+        
+    } catch (error) {
+      return false;
```
Comment (joanna209) [inline src/contact/contactForm.js:29]: Console log the error here, with console.error(error). It's hard to tell what happened if we only return false here

**Review by joanna209 — CHANGES_REQUESTED** [review]

File: src/contact/contactForm.js (hunk 2 above)
Comment (mrzpn) [reply src/contact/contactForm.js:18]: I've tried implementing it this way, but I couldn't get it working. I need the async/await method to perform the captcha verification and I don't believe this works with .then() syntax (unless I'm missing something?)

Comment (joanna209) [reply src/contact/contactForm.js:18]: Ah that's alright - thanks for trying it out anyways

**Review by joanna209 — APPROVED** [review]

**Pull request merged**